ql/src/test/queries/clientpositive/filter_emptyliteral.q

```diff
@@ -0,0 +1,3 @@
+create table employee(name varchar(255), id int);
+
+explain cbo SELECT id FROM employee where name = '';
```

Comment:
Please add another test for `char` for completeness. Otherwise, looks good to me!

Reply:
I have added this. Also I have removed the redundant piece of code for CHAR and VARCHAR sql type.